core/state/statedb.go

```diff
@@ -831,13 +830,14 @@ func (s *StateDB) SetCode(addr common.Address, code []byte) {
 	}
 }
 
-func (s *StateDB) SetState(addr common.Address, key, value common.Hash) {
+func (s *StateDB) SetState(addr common.Address, key, value common.Hash) common.Hash {
 	stateObject := s.getOrNewStateObject(addr)
 	if stateObject != nil {
 		stateObject = s.mvRecordWritten(stateObject)
-		stateObject.SetState(key, value)
 		MVWrite(s, blockstm.NewStateKey(addr, key))
+		return stateObject.SetState(key, value)
```

Comment:
@cffls earlier the order was to set state and then do MVWrite. Because we want to return the new value, I've changed the order. I hope this doesn't affect anything.